Automattic/simplenote-macos
# Notes List: New Sort Modes

### Fix
In this PR we're implementing new Sort Modes in the Sort List, so that we match iOS, Electron and Android:

- Alphabetically: A-Z / Z-A
- Creation Date: Newer / Older
- Modification Date: Newer / Older

@SylvesterWilmott Matching Electron's Menu. WDYT? (Thanks in advance!!)

cc @eshurakov (Thankyooou)

Closes #33
Ref. #626

### Test: Migration / Updated
1. Install **develop** in your mac and authenticate
2. Click over `View > Note Sorting > Date Updated`
3. Install this branch on top

- [x] Verify `Note Sorting > Modified: Newest` is preselected for you

### Test: Migration / Alpha
1. Run this in bash
  ```defaults delete com.automattic.SimplenoteMac.beta notesListSortMode
   defaults write com.automattic.SimplenoteMac.beta kAlphabeticalSortPreferencesKey 1
  ```
2. Run this branch on top

 - [x] Verify `Note Sorting > Alphabetically: A-Z` is preselected

### Test: Sort Modes
1. Run this branch in your mac
2. Launch iOS as well!

- [x] Verify the setting `Note Sorting > Alphabetically: A-Z` matches iOS
- [x] Verify the setting `Note Sorting > Alphabetically: Z-A` matches iOS
- [x] Verify the setting `Note Sorting > Created: Newest` matches iOS
- [x] Verify the setting `Note Sorting > Created: Oldest` matches iOS
- [x] Verify the setting `Note Sorting > Modified: Newest` matches iOS
- [x] Verify the setting `Note Sorting > Modified: Oldest` matches iOS

### Release
`RELEASE-NOTES.txt` was updated in 536257b with:
 
> New Notes List Sort Modes #33

### Screenshots
Before|Now
-|-
<img width="313" alt="Screen Shot 2020-12-14 at 5 31 09 PM" src="https://user-images.githubusercontent.com/1195260/102133198-7f9a3980-3e33-11eb-8cb4-c394eacaa5b4.png">|<img width="343" alt="Screen Shot 2020-12-15 at 12 20 01 PM" src="https://user-images.githubusercontent.com/1195260/102234593-351aca80-3ed0-11eb-8630-bd0d4c594c09.png">
